### ISSUE TITLE
iOS crash

### ISSUE THREAD

**Issue by cristianoccazinsp** [realm/realm-js#2828]
## Goals
Random crash from our crash reporting tool on an iPhone XR. Stack trace is as follows:
```
EXC_BAD_ACCESS Attempted to dereference garbage pointer 0x10. 
    Frameworks/JavaScriptCore.framework/JavaScriptCore JSC::JSGlobalObject::getOwnPropertySlot(JSC::JSObject*, JSC::ExecState*, JSC::PropertyName, JSC::PropertySlot&)
    Frameworks/JavaScriptCore.framework/JavaScriptCore JSC::JSProxy::getOwnPropertySlot(JSC::JSObject*, JSC::ExecState*, JSC::PropertyName, JSC::PropertySlot&)
    Frameworks/JavaScriptCore.framework/JavaScriptCore JSC::JSObject::get(JSC::ExecState*, JSC::PropertyName) const
    Frameworks/JavaScriptCore.framework/JavaScriptCore _JSObjectGetProperty
    zinspector3 void std::__1::__call_once_proxy<std::__1::tuple<realm::SyncManager& realm::js::syncManagerShared<realm::jsc::Types>(realm::jsc::Types::Context&)::{lambda()#1}&&> >(void*)
    zinspector3 void std::__1::__call_once_proxy<std::__1::tuple<realm::SyncManager& realm::js::syncManagerShared<realm::jsc::Types>(realm::jsc::Types::Context&)::{lambda()#1}&&> >(void*)
    zinspector3 bool realm::js::wrap<&realm::js::ListClass<realm::jsc::Types>::set_index>(OpaqueJSContext const*, OpaqueJSValue*, unsigned int, OpaqueJSValue const*, OpaqueJSValue const**)
    zinspector3 OpaqueJSValue const* realm::js::wrap<&realm::js::ResultsClass<realm::jsc::Types>::get_index>(OpaqueJSContext const*, OpaqueJSValue*, unsigned int, OpaqueJSValue const**)
    zinspector3 OpaqueJSValue const* realm::js::wrap<&realm::js::ResultsClass<realm::jsc::Types>::sorted>(OpaqueJSContext const*, OpaqueJSValue*, OpaqueJSValue*, unsigned long, OpaqueJSValue const* const*, OpaqueJSValue const**)
    Frameworks/JavaScriptCore.framework/JavaScriptCore long long JSC::APICallbackFunction::call<JSC::JSCallbackFunction>(JSC::ExecState*)
    Frameworks/JavaScriptCore.framework/JavaScriptCore _llint_entry
    Frameworks/JavaScriptCore.framework/JavaScriptCore _llint_entry
    Frameworks/JavaScriptCore.framework/JavaScriptCore _llint_entry
    Frameworks/JavaScriptCore.framework/JavaScriptCore _llint_entry
    Frameworks/JavaScriptCore.framework/JavaScriptCore _llint_entry
    Frameworks/JavaScriptCore.framework/JavaScriptCore _llint_entry
    Frameworks/JavaScriptCore.framework/JavaScriptCore _llint_entry
    Frameworks/JavaScriptCore.framework/JavaScriptCore _llint_entry
    Frameworks/JavaScriptCore.framework/JavaScriptCore _llint_entry
    Frameworks/JavaScriptCore.framework/JavaScriptCore _vmEntryToJavaScript
    Frameworks/JavaScriptCore.framework/JavaScriptCore JSC::Interpreter::executeCall(JSC::ExecState*, JSC::JSObject*, JSC::CallType, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&)
    Frameworks/JavaScriptCore.framework/JavaScriptCore JSC::boundThisNoArgsFunctionCall(JSC::ExecState*)
    Frameworks/JavaScriptCore.framework/JavaScriptCore _vmEntryToNative
    Frameworks/JavaScriptCore.framework/JavaScriptCore JSC::Interpreter::executeCall(JSC::ExecState*, JSC::JSObject*, JSC::CallType, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&)
    Frameworks/JavaScriptCore.framework/JavaScriptCore JSC::profiledCall(JSC::ExecState*, JSC::ProfilingReason, JSC::JSValue, JSC::CallType, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&)
    Frameworks/JavaScriptCore.framework/JavaScriptCore _JSObjectCallAsFunction
    zinspector3 facebook::react::JSIExecutor::defaultTimeoutInvoker(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>)
    zinspector3 facebook::react::JSIExecutor::defaultTimeoutInvoker(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>)
    zinspector3 facebook::react::JSIExecutor::defaultTimeoutInvoker(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>)
    zinspector3 facebook::react::JSIExecutor::defaultTimeoutInvoker(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>)
    zinspector3 facebook::react::JSIExecutor::defaultTimeoutInvoker(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>)
    zinspector3 facebook::react::JSIExecutor::defaultTimeoutInvoker(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>)
    zinspector3 facebook::react::JSIExecutor::defaultTimeoutInvoker(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>)
    zinspector3 facebook::react::JSIExecutor::defaultTimeoutInvoker(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>)
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFRunLoopDoBlocks
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFRunLoopRun
    Frameworks/CoreFoundation.framework/CoreFoundation _CFRunLoopRunSpecific
    zinspector3 facebook::react::JSIExecutor::defaultTimeoutInvoker(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>)
    Frameworks/Foundation.framework/Foundation -[NSThread main]
    Frameworks/Foundation.framework/Foundation ___NSThread__start__
    /usr/lib/system/libsystem_pthread.dylib __pthread_start
```

## Version of Realm and Tooling

- Realm JS SDK Version: 5.0.3
- Node or React Native: React Native 0.61.5
- Client OS & Version: iOS 13.3.1

**Comment by thomasarogers** [comment]
Same issue

**Comment by vineetshetty** [comment]
I have consistently seen this issue crop up in various versions of realmjs. I can confirm that this issue exists right from 5.0.1 through 5.0.3

This issue does not always come up but randomly affects our users. There does not seem to be a pattern with this issue.

**Comment by cristianoccazinsp** [comment]
Some more stack traces / similar crashes here: https://github.com/realm/realm-js/issues/2853

The only pattern I've seen so far is that most of the crashes happen when the app goes to background.

**Comment by jdmunro** [comment]
Also exists in 6.x

**Comment by oojikoo** [comment]
I face this issue quite a lot in version 6.x. what should we do with this issue really?

**Comment by kneth** [comment]
Have you observed similar crashes with version 10.x?

**Comment by kneth** [comment]
We have released v11.0.0 which has a complete rewrite of the integration with JSC. I am closing the issue, and ask you to upgrade. If the issue is still observed, please create a new issue.
